pkg/model/wasm_engine.go

```diff
@@ -0,0 +1,95 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+const (
+	EngineTypeWasm                    = "wasm"
+	EngineKeyEntryModuleWasm          = "entrymodule"
+	EngineKeyEntrypointWasm           = "entrypoint"
+	EngineKeyParametersWasm           = "parameters"
+	EngineKeyEnvironmentVariablesWasm = "environmentvariables"
+	EngineKeyImportModulesWasm        = "importmodules"
+)
+
+// NewWasmEngineSpec returns an EngineSpec of type EngineTypeWasm with the provided arguments as EngineSpec.Params.
+func NewWasmEngineSpec(
+	entryModule StorageSpec,
+	entrypoint string,
+	parameters []string,
+	environmentVariables map[string]string,
```

Comment:
The environmentVariables here are a map of string->string, but the environmentVariables in the docker engine are []string.  Should we fix them so they are both consistent?